src/core/filter/security.go

```diff
@@ -289,7 +279,11 @@ func (it *idTokenReqCtxModifier) Modify(ctx *beegoctx.Context) bool {
 		log.Warning("User matches token's claims is not onboarded.")
 		return false
 	}
-	groupNames := oidc.GroupsFromToken(claims)
+	settings, err := config.OIDCSetting()
+	if err != nil {
+		log.Errorf("Failed to get OIDC settings, error: %v", err)
+	}
+	groupNames, _ := oidc.GroupsFromClaims(claims, settings.GroupsClaim)
```

Comment:
If groupNames, ok := oidc.GroupsFromClaims(claims, settings.GroupClaim); ok {
...
}
because it is useless to process an empty groupNames

Reply:
This is trivial, but I'll make the change,.